docs/docs/topics/mutual-auth.md

```diff
@@ -0,0 +1,390 @@
+
+## JWT Verification: Application-based Mutual Authentication
+
+> - **Security**: Strong
+> - **Difficulty**: Moderate
+> - **Encrypted**: Yes (from the end user to Pomerium. It's up to the service to provide a TLS endpoint for Pomerium to use.)
+> - **Mutual Authentication**: Application Layer
+
+Many, but not all, modern web applications support [json web tokens][jwt-rfc] (**JWTs**). These tokens can be provided by Pomerium (with the [`pass_identity_headers`] key) so that an upstream service can independently verify that the traffic has properly gone through Pomerium, and that the incoming user request is authorized.
+
+JWTs are verified the same way as TLS certificates, by checking that they are signed by a trusted authority. Pomerium provides that authority with a json web key set (**JWKS**) which signs the JWT. Upstream services that utilize JWTs can be directed to Pomerium for the JWKS.
+
+Let's look at an example:
+
+```mermaid
+flowchart LR
+  D(End User)
+  C(Identity Provider)
+  D==oA
+  subgraph lan[Internal Network]
+    direction LR
+    style lan stroke-dasharray: 5 5
+    A(Pomerium)
+    B(Upstream Service)
+    A ==o B
+  end
+  B-.JWKS Verification.-A
+  C-.-A
+```
+
+1. When the user connects to the route `service.example.com` the traffic is received Pomerium.
+1. Pomerium redirects the user to the identity provider (**IdP**) to sign in and validate their identity.
+1. After authentication succeeds, Pomerium verifies that the authenticated user is authorized to access the upstream service, and sends the connection to it, including the `X-Pomerium-JWT-Assertion` header containing the JWT signed by the key set in Pomerium.
+1. The upstream service reads the signing key from Pomerium, and uses it to validate the JWT. Finding that the JWT was signed by the trusted key, the traffic is allowed.
+
+Our hacker may be able to forge a basic JWT, but they can't sign it with the key secured in the Pomerium configuration, so the upstream service rejects their connections:
+
+```mermaid
+flowchart LR
+D(End User)
+C(Identity Provider)
+D==oA
+subgraph lan[Internal Network]
+  direction LR
+  style lan stroke-dasharray: 5 5
+  A(Pomerium)
+  B(Upstream Service)
+  A ==o B
+end
+B-.JWKS Verification.-A
+C-.-A
+E[/Hacker/] --x B
+```
+
+In this way, we've applied a zero-trust security model to the application layer of our infrastructure's network model. You can see JWT verification in practice with our [Grafana] integration guide.
+
+## mTLS: Protocol-based Mutual Authentication
+
+> - **Security**: Excellent
+> - **Difficulty**: Hard
+> - **Encrypted**: Yes
+> - **Mutual Authentication**: Protocol Layer
+
+Most tech professionals are familiar with [Transport Layer Security] (**TLS**). The majority of traffic on the web today is sent using TLS. In addition to encrypting data using the server's TLS certificate, the server identity is validated by the certificate and the Certificate Authority (**CA**) that signed it.
```

Comment:
Maybe we should add a subheading clarifying mTLS between browser and pomerium and pomerium and upstreams?

Reply:
Resolved by review changes.